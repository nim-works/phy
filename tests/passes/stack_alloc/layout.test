discard """
  description: "
    Ensure that the locations have the correct alignment and layout
  "
  output: "(Done)"
"""
(TypeDefs
  (Blob 1 1)
  (Blob 1 2)
  (Blob 1 4)
  (Blob 1 8)
  (Blob 17 4)
  (ProcTy (Void)))
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 5)
    (Locals (Type 4) (Type 3) (Type 0) (Type 2) (Type 0) (Type 1))
    (List
      (Block (Params)
        (Drop (Addr (Local 0)))
        (Drop (Addr (Local 1)))
        (Drop (Addr (Local 2)))
        (Drop (Addr (Local 3)))
        (Drop (Addr (Local 4)))
        (Drop (Addr (Local 5)))
        (Return)))))
