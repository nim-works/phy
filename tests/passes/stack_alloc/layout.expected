;$sexp
(TypeDefs
  (Blob 1 1)
  (Blob 1 2)
  (Blob 1 4)
  (Blob 1 8)
  (Blob 17 4)
  (ProcTy (Void)))
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 5) 32
    (Locals (UInt 8))
    (List
      (Block
        (Params (Local 0))
        (Drop
          (Copy (Local 0)))
        (Drop
          (Add (UInt 8)
            (Copy (Local 0))
            (IntVal 24)))
        (Drop
          (Add (UInt 8)
            (Copy (Local 0))
            (IntVal 25)))
        (Drop
          (Add (UInt 8)
            (Copy (Local 0))
            (IntVal 28)))
        (Drop
          (Add (UInt 8)
            (Copy (Local 0))
            (IntVal 29)))
        (Drop
          (Add (UInt 8)
            (Copy (Local 0))
            (IntVal 30)))
        (Return)))))
